### PR TITLE
Add Bucket and ObjecTMD models in Arsenal

### PR DESCRIPTION
**Pull request by jeremyds**
This two models are needed for the S3SOFS feature into the cdmiclient
repository. The S3 one does not export modules and shared ones are in
Arsenal so move these two model modules to it.

### CODE REVIEW

File: lib/models/BucketInfo.js [hunk 1]
```diff
@@ -0,0 +1,503 @@
+const assert = require('assert');
+const { WebsiteConfiguration } = require('./WebsiteConfiguration');
```
Comment (jeremyds) [inline lib/models/BucketInfo.js]: should be moved too :(

File: lib/models/BucketInfo.js [hunk 2]
```diff
@@ -0,0 +1,503 @@
...
+const ReplicationConfiguration =
```
Comment (jeremyds) [inline lib/models/BucketInfo.js]: same here

File: lib/models/BucketInfo.js (hunk 1 above)
Comment (jeremyds) [inline lib/models/BucketInfo.js]: done

File: lib/models/BucketInfo.js (hunk 2 above)
Comment (jeremyds) [inline lib/models/BucketInfo.js]: done

**Comment by rahulreddy** [comment]
We need some units tests to go along with this and a PR in S3 removing the models and using them from Arsenal.

**Comment by jeremyds** [comment]
@rahulreddy The related S3 PR is here: https://github.com/scality/S3/pull/820
What kind of tests do you think about to test a code move?
I guessed these source files are used and tested in the S3 repository but if you think more tests may help to check the code move do not hesitate ;)

**Comment by rahulreddy** [comment]
No, I mean generally when we move these files from S3 we move the accompanying tests as well. If there were none, we add some. Otherwise it would not make sense to have code in the project that is not tested.

**Comment by rahulreddy** [comment]
@jeremyds This PR is missing some property changes that were made for backbeat replication. I was going to make the changes manually. Please reach out to me next time before merging manually. We usually need two approvals and run an end to end before merging.

**Pull request merged**